CARDS-1497: Terms of Use on patient's first login

veronikaslc opened the pull request:
[CARDS-1497](https://phenotips.atlassian.net/browse/CARDS-1497) - _Terms of Use on patient's first login_
[CARDS-1576](https://phenotips.atlassian.net/browse/CARDS-1576) - _If the terms of use change, patient users should be prompted to review and re-accept_

This PR also includes: 
* [CARDS-1573](https://phenotips.atlassian.net/browse/CARDS-1573) - _Better alignment on the patient survey page_
* [CARDS-1574](https://phenotips.atlassian.net/browse/CARDS-1574) - _Patient UI overhaul_ (partial):
  * Better looking appbar
  * Fixed some (but not all) JS console warnings (missing `key`)
  * Improved the look of the welcome page (right after patient login)
  * Better positioning of Form header (fixing layout issue caused by introduction of Footer for 1497)

### Testing instructions
The implementation already assumes that the "Patient infomation" have already been created before the patient logs in to fill out their surveys (as will happen in production).

Before testing the Terms of Use workflow, **pre-create the relevant subjects and forms by importing
[testPatients.csv](https://github.com/data-team-uhn/cards/files/7834492/testPatients.csv) and [testVisits.csv](https://github.com/data-team-uhn/cards/files/7834491/testVisits.csv)**, which mirror the known test patients:

```
curl -u admin:admin "localhost:8080/Forms/" -X POST -F ":data=@testVisits.csv" -F":questionnaire=/Questionnaires/Visit information" -F ":subjectType=/SubjectTypes/Patient" -F ":subjectType=/SubjectTypes/Patient/Visit"
```

```
curl -u admin:admin "localhost:8080/Forms/" -X POST -F ":data=@testPatients.csv" -F":questionnaire=/Questionnaires/Patient information" -F ":subjectType=/SubjectTypes/Patient"
```
### To do

[CARDS-1577](https://phenotips.atlassian.net/browse/CARDS-1577) - Update the text of the Terms of Use
- Final review of the Terms of Use text: ensure consistent app name (EPRO vs DATAPRO vs DATA PRO), correct addresses, etc.

Commit:
CARDS-1497 / CARDS-1498: Add a field tou_accepted:boolean in the Patient information form

Commit:
CARDS-1497 / CARDS-1554: Display ToU based on the tou_accepted field

Commit:
CARDS-1497 / CARDS-1555: Add a link to ToU for identified patient users in the proms UI

sdumitriu reviewed (dismissed):
```
Error: Can't resolve './TOU.json'
```

marta- commented on proms-resources/clinical-data/src/main/resources/SLING-INF/content/Questionnaires/Patient information.xml:
```diff
@@ -178,4 +178,23 @@
+		<primaryNodeType>cards:Question</primaryNodeType>
+		<property>
+			<name>text</name>
+			<value>The patient consents to terms of use</value>
```
```suggestion
			<value>The patient accepted the Terms of Use</value>
```

marta- commented on proms-resources/frontend/src/main/frontend/src/MockPatientIdentification.jsx:
```diff
@@ -131,12 +145,11 @@ function MockPatientIdentification(props) {
   }
 
-  // When the visit is successfully obtained, pass it along with the identification data
+  // When the visit is successfully obtained ant terms of use accepted, pass it along with the identification data
```
```suggestion
  // When the visit is successfully obtained and Terms of Use accepted, pass it along with the identification data
```

marta- commented on proms-resources/frontend/src/main/frontend/src/MockPatientIdentification.jsx:
```diff
@@ -247,10 +259,46 @@ function MockPatientIdentification(props) {
+  // Show terms of use dialog
+  useEffect(() => {
+    if (showTou) {
+      let tou = require('./TOU.json');
```
Let's rename the json:
```suggestion
      let tou = require('./ToU.json');
```

Commit:
CARDS-1497: Added missing file

marta- commented on proms-resources/frontend/src/main/frontend/src/MockPatientIdentification.jsx:
```diff
@@ -280,7 +328,33 @@ function MockPatientIdentification(props) {
+         Decline
+       </Button>
+     </DialogActions>
+   </ResponsiveDialog>
```
IMO the entire UI for the Terms of Use (and loading the text from the Json) dialog should go in a separate component.

marta- replied:
The move should include the confirmation dialog and the logic around it.

marta- replied:
Done in 5e0ba46

marta- commented on proms-resources/frontend/src/main/frontend/src/TOU.json:
```diff
@@ -0,0 +1,3 @@
+{
+  "text" : "Licensed to the **CARDS** Apache Software Foundation (ASF) under one or more contributor license agreements. See the NOTICE file  distributed with this work for additional information regarding copyright ownership. The ASF licenses this file to you under the Apache License, Version 2.0 (the 'License'); you may not use this file except in compliance with the License. You may obtain a c ...
```
The Terms of Use text has been attached to the Jira task by @k-pace. It will have to be formatted using markdown.

marta- commented on proms-resources/frontend/src/main/frontend/src/index.jsx:
```diff
@@ -68,7 +79,19 @@ function PromsHomepage (props) {
     <AppBar position="static" className={classes.appbar}>
       <Toolbar variant="dense" className={classes.toolbar}>
-        <img src="/libs/cards/resources/logo.png" alt="logo" className={classes.logo} />
+        <div>
+          <img src="/libs/cards/resources/logo.png" alt="logo" className={classes.logo} />
+          <Link
+            component="button"
+            color="inherit"
+            variant="body2"
+            underline="always"
+            onClick={() => {setShowTou(true);}}
+            className={classes.tou}
+          >
+            Terms of Use
+          </Link>
+        </div>
```
The layout doesn't look great. Let's make "Terms of Use" centered in the header. IIRC the toolbar should already have flex display and removing the wrapping div should be enough for the change below (removing the wrapper div) to work. Also please remove the underline.
```suggestion
        <img src="/libs/cards/resources/logo.png" alt="logo" className={classes.logo} />
        <Link
          component="button"
          color="inherit"
          variant="body2"
          underline="none"
          onClick={() => {setShowTou(true);}}
          className={classes.tou}
        >
          Terms of Use
        </Link>
```

![image](https://user-images.githubusercontent.com/651980/148438348-f97f83dc-712e-4831-abf9-6b528d2e8945.png)

marta- commented on proms-resources/frontend/src/main/frontend/src/index.jsx:
```diff
@@ -77,6 +100,17 @@ function PromsHomepage (props) {
+      onClose={() => {setShowTou(false);}}
+    >
+     <DialogContent dividers>
+       <FormattedText variant="caption">{tou.text}</FormattedText>
```
`caption` is too small, let's use the default instead.
```suggestion
       <FormattedText>{tou.text}</FormattedText>
```

marta- commented on proms-resources/frontend/src/main/frontend/src/MockPatientIdentification.jsx:
```diff
@@ -280,7 +328,33 @@ function MockPatientIdentification(props) {
+  return (<>
+   <ResponsiveDialog title="Terms of Use" open={showTou} width="lg" >
+     <DialogContent dividers>
+       <FormattedText variant="caption">{touText}</FormattedText>
```
`caption` is too small, let's use the default instead.
```suggestion
       <FormattedText>{tou.text}</FormattedText>
```

marta- commented on proms-resources/frontend/src/main/frontend/src/TOU.json:
```diff
@@ -0,0 +1,3 @@
+{
```
The JSON should also specify the title to be set to the dialog. Concretely, per the file attached to the jira task, it should be 
```
  "title" : "EPRO Terms of Use",
```

Commit:
CARDS-1497: Terms of Use on patient's first login
Addressed codereview comments

marta- commented on proms-resources/frontend/src/main/frontend/src/TOU.json:
```diff
@@ -0,0 +1,4 @@
+{
+  "title" : "EPRO Terms of Use",
+  "text" : "These terms and conditions (the 'Agreement') for a symptoms survey located at ePRO.uhn.ca and any associated website, portal, or similar service (collectively, the 'Application'), constitute a legal agreement and is entered into by and between you ('you', 'your') and the University Health Network ('UHN', 'we', 'us', 'our').<br><br>**IMPORTANT! BY USING OR ACCESSING THE APPLICATION, YO ...
```
HTML formatting should be avoided, let's stick to pure markdown.

Please replace `'` with `\"`.
`h5` and `h6` for subheadings appear too small. Let's use `h3` and `h4` instead. (`### Subtitle`, `#### Subtitle`).
New lines can be marked with `\\\n`, but to make the sources more readable I suggest making `text` an array of lines and joining them on the frontend. E.g.:
```
  "text": [
    "These terms and conditions (the \"Agreement\") for a symptoms survey located at ePRO.uhn.ca and any associated website, portal, or similar service (collectively, the \"Application\"), constitute a legal agreement and is entered into by and between you (\"you\", \"your\") and the University Health Network (\"UHN\", \"we\", \"us\", \"our\").",
    "**IMPORTANT! BY USING OR ACCESSING THE APPLICATION, YOU AGREE TO BECOME BOUND BY THE AGREEMENT. IF YOU DO NOT AGREE WITH THE AGREEMENT, YOU MUST NOT USE OR ACCESS THE APPLICATION. NOTE THAT THERE ARE ALTERNATIVE METHODS TO COMPLETE THE SURVEY IF YOU WOULD NOT LIKE TO PROCEED WITH THIS APPLICATION.**",
    "This Agreement governs your access to the Application and use, including any content, functionality, and services offered on or through the Application.",
    "The Application is intended to act as an information collection tool for physicians to better prepare for appointments with patients and to improve record keeping. The Application is intended to facilitate symptom reporting to healthcare providers, for quality monitoring and improvement, and general patient education relative to cancer symptom status.",
    ....
  ]
```

marta- reviewed (changes requested):
Please rename the file "TOU.json" to "ToU.json".

Commit:
CARDS-1497: Terms of Use on patient's first login
Used markdown formatting

marta- commented on proms-resources/frontend/src/main/frontend/src/TOU.json:
```diff
@@ -1,4 +1,80 @@
   "title" : "EPRO Terms of Use",
-  "text" : "These terms and conditions (the 'Agreement') for a symptoms survey located at ePRO.uhn.ca and any associated website, portal, or similar service (collectively, the 'Application'), constitute a legal agreement and is entered into by and between you ('you', 'your') and the University Health Network ('UHN', 'we', 'us', 'our').<br><br>**IMPORTANT! BY USING OR ACCESSING THE APPLICATION, YO ...
+  "text" : [
+    "These terms and conditions (the \"Agreement\") for a symptoms survey located at ePRO.uhn.ca and any associated website, portal, or similar service (collectively, the \"Application\"), constitute a legal agreement and is entered into by and between you (\"you\", \"your\") and the University Health Network (\"UHN\", \"we\", \"us\", \"our\").\\\n",
```
There should be no need for newline markers at the end of each array item, since we join by newline in the code. We can simply join by double newline instead.
```suggestion
    "These terms and conditions (the \"Agreement\") for a symptoms survey located at ePRO.uhn.ca and any associated website, portal, or similar service (collectively, the \"Application\"), constitute a legal agreement and is entered into by and between you (\"you\", \"your\") and the University Health Network (\"UHN\", \"we\", \"us\", \"our\").",
```

marta- replied:
Done in 8c43d6e

marta- commented on proms-resources/frontend/src/main/frontend/src/TOU.json:
```diff
@@ -1,4 +1,80 @@
+    "**IMPORTANT! BY USING OR ACCESSING THE APPLICATION, YOU AGREE TO BECOME BOUND BY THE AGREEMENT. IF YOU DO NOT AGREE WITH THE AGREEMENT, YOU MUST NOT USE OR ACCESS THE APPLICATION. NOTE THAT THERE ARE ALTERNATIVE METHODS TO COMPLETE THE SURVEY IF YOU WOULD NOT LIKE TO PROCEED WITH THIS APPLICATION.**\\\n",
+    "This Agreement governs your access to the Application and use, including any content, functionality, and services offered on or through the Application.\\\n",
+    "The Application is intended to act as an information collection tool for physicians to better prepare for appointments with patients and to improve record keeping. The Application is intended to facilitate symptom reporting to healthcare providers, for quality monitoring and improvement, and general patient education relative to cancer symptom status. It is not a medical device. Nothing cont ...
+    "####Not Medical Advice.\\\n",
```
The headings don't get rendered because the correct syntax is `#### Title`, with space between the heading marker `####` and the text.
```suggestion
    "#### Not Medical Advice.\\\n",
```

marta- replied:
Fixed in 8c43d6e

marta- commented on proms-resources/frontend/src/main/frontend/src/ToUDialog.jsx:
```diff
@@ -0,0 +1,81 @@
+      onClose={onClose}
+    >
+      <DialogContent dividers>
+       <FormattedText>{tou.text.join('\\\n')}</FormattedText>
```
Looks like `\` to mark newlines doesn't work as expected (generates the `\` character in the output). Let's use the other way of marking newlines, double space followed by `\n`. Additionally, let's double the newline to compensate for the markers removed from each line in the sources.
```suggestion
       <FormattedText>{tou.text.join('  \n\n')}</FormattedText>
```

marta- replied:
Done in 8c43d6e

marta- commented on proms-resources/frontend/src/main/frontend/src/TOU.json:
```diff
@@ -0,0 +1,80 @@
+    "Only aggregated, anonymized data is periodically transmitted to external services to help us improve the Application and our service. We will share your information with third parties only in the ways that are described in this privacy statement. We may disclose User Provided and Automatically Collected Information: * As required by law, such as to comply with a subpoena, or similar legal pr ...
+    "####Use of your Information by your Health Care Team\\\n",
+    "The information you enter into this Application will be provided to your healthcare team and will become part of your medical record.\\\n",
+    "####What are my rights**\\\n",
```
There was supposed to be a question mark at the end of this line, that probably got confused with a special character and got replaced with `**` in the formatting process. There are several other occurrences of this replacement across the text.
```suggestion
    "####What are my rights?",
```

marta- replied:
Fixed in 8c43d6e

marta- commented on proms-resources/frontend/src/main/frontend/src/TOU.json:
```diff
@@ -0,0 +1,80 @@
+    "Email: privacy@uhn.ca",
+    "Website: http://www.uhn.ca/PatientsFamilies/Patient_Safety_Advocacy/Privacy/Pages/privacy_policy.aspx \\\n",
+    "####Complaints\\\n",
+    "If you have complaints about UHNs compliance with this privacy policy or PHIPA, please contact UHNs Privacy Office at the above contact information. Further questions and concerns can be directed to the Office of the Information and Privacy Commissioner of Ontario at:",
```
Some apostrophes that were in the original text are missing here.

```suggestion
    "If you have complaints about UHN's compliance with this privacy policy or PHIPA, please contact UHN's Privacy Office at the above contact information. Further questions and concerns can be directed to the Office of the Information and Privacy Commissioner of Ontario at:",
```

marta- replied:
Fixed in 8c43d6e

marta- commented on proms-resources/frontend/src/main/frontend/src/TOU.json:
```diff
@@ -0,0 +1,80 @@
+    "####Not Medical Advice.\\\n",
+    "The App is not a substitute for professional medical advice, diagnosis or treatment. Never make changes to your medication, nor adjust your dose, without first consulting your health care provider. Always seek the advice of a physician or other qualified healthcare provider concerning questions you have regarding a medical condition, and before starting, stopping or modifying any treatment o ...
+    "####No Regulatory Approval.\\\n",
+    "The App has not been evaluated by Health Canada, the United States Food and Drug Administration (**USFDA**) or any other domestic or foreign regulatory agency and is not approved to diagnose, treat, cure or prevent any disease or for the prescription of any medication or treatment. While the information contained may be of use in other jurisdictions, these materials were provided and designe ...
```
According to the original text, these look like they're supposed to be paragraph titles, on the same line.
```suggestion
    "**No Regulatory Approval.** The App has not been evaluated by Health Canada, the United States Food and Drug Administration (**USFDA**) or any other domestic or foreign regulatory agency and is not approved to diagnose, treat, cure or prevent any disease or for the prescription of any medication or treatment. While the information contained may be of use in other jurisdictions, these materials were provided and designed for use in Canada and you are solely responsible for ensuring suitability and permitted use in other jurisdictions.\\\n",
```

marta- replied:
Done in 8c43d6e

marta- committed:
CARDS-1497: Terms of Use on patient's first login
Renamed the file to match the request call

marta- committed:
CARDS-1497: Terms of Use on patient's first login
* Fixed markdown syntax to render headings
* Fixed newline syntax to not output the `\` character
* Re-added missing `'` and `?` into the text
* Improved the display of some nested subheadings
* Improved the display of contact information

marta- reviewed (commented):
The build is still failing because TOU.json hasn't yet been renamed to ToU.json (lowercase o).

marta- commented:
> The build is still failing because TOU.json hasn't yet been renamed to ToU.json (lowercase o).

Fixed in eb2ca9b

veronikaslc commented:
> > The build is still failing because TOU.json hasn't yet been renamed to ToU.json (lowercase o).
> 
> Fixed in [eb2ca9b](https://github.com/data-team-uhn/cards/commit/eb2ca9bb0e5accbdd072881de8fcd4ae387f8fa6)

Strange that apparently git does not recognize the renaming of file if the change is in between lower and upper cases so my commit did not catch it. File system has the change, git client did not see it and build runs OK.

marta- committed:
CARDS-1497: Terms of Use on patient's first login
* Refactored the code to move the action buttons and confirmation dialog inside the ToUDialog component
* Replaced the X close button with an action button at the bottom for consistency

marta- committed:
[misc] Cleaned up the indentation, removed unused import

marta- commented on proms-resources/frontend/src/main/frontend/src/MockPatientIdentification.jsx:
```diff
@@ -247,10 +254,27 @@ function MockPatientIdentification(props) {
+    }
+  }
+
+  // Show terms of use dialog
+  useEffect(() => {
+    if (showTou) {
+      setShowTou(true);
+    }
+  }, [showTou]);
```
What is the purpose of this effect?

veronikaslc replied:
Artifact leftover after some refactoring, removed in 5ac1b1f.

Commit:
CARDS-1497: Terms of Use on patient's first login
Removed acritfact effect

k-pace commented:
Minor but can we remove the inline quote styling and just plain indent for the address? I'll note that this is definitely not the correct address so will provide update as soon as I have it

![image](https://user-images.githubusercontent.com/33154316/148663402-80c1c73e-b1f9-40b0-b826-bbc5bc4a61db.png)

k-pace commented:
Actually, looked at each of the addresses; I think it looks good (as does everything else with regards to styling)

marta- committed:
CARDS-1497: Terms of Use on patient's first login
Addressed review comments: adjusted the style of addresses

marta- committed:
CARDS-1497: Terms of Use on patient's first login
Bug fix: confirmation dialog doesn't disappear after clicking its Decline button

marta- committed:
CARDS-1497: Terms of Use on patient's first login
Fixed markdown formatting for generating itemized list

marta- commented:
> Minor but can we remove the inline quote styling and just plain indent for the address? 

@k-pace Done in 95c8bf7, it looks like in the screenshot below. It's easy to revert if it's not a true improvement.

![image](https://user-images.githubusercontent.com/651980/148664492-cac48567-1284-4d3e-91ba-204e7f861a16.png)

marta- committed:
CARDS-1497: Terms of Use on patient's first login
Refactoring: decouple the code responsible for handling Terms of Use from placeholder code that currently ensures patient info synchronization, to make the latter easier to remove when real patient identification is implemented.
Bug fix: tou_accepted is now correctly saved in Patient information upon acceptance by patient, and coorectly read in subsequent sessions

marta- committed:
CARDS-1497: Terms of Use on patient's first login
Refactoring: moved the terms of use text to a JCR node, to make it easier to update in production

marta- commented:
> ### To do
> *  Move the terms of use from sources to a JCR node, for easier updates

Done in 9a3118f

sashaandjic commented:
<img width="395" alt="image" src="https://user-images.githubusercontent.com/23243244/148746526-5a4571c5-9a1a-4477-a97b-3cc496b48b66.png">

<img width="456" alt="image" src="https://user-images.githubusercontent.com/23243244/148746605-371f7622-568c-4c94-97c1-d395714b545c.png">

<img width="392" alt="image" src="https://user-images.githubusercontent.com/23243244/148747091-443664a3-cc2e-47cd-aba6-11410a6d62dd.png">

sashaandjic commented:
Checked for consistent app name in the text. Only EPRO is used.

sashaandjic commented:
QUESTION: should the yellow highlighted text be bolded? If so, should blue highlighted "the" be removed?
<img width="365" alt="image" src="https://user-images.githubusercontent.com/23243244/148751362-e2ff42ec-9a8e-473e-8930-f0955905ff55.png">

k-pace committed:
CARDS-1497: Terms of Use on patient's first login
Changed ePRO to DATA-PRO

sashaandjic reviewed (approved):
tested
[screencapture-localhost-1497-Proms-html-Cardio-2022-01-14-14_47_55.pdf](https://github.com/data-team-uhn/cards/files/7872829/screencapture-localhost-1497-Proms-html-Cardio-2022-01-14-14_47_55.pdf)

marta- committed:
CARDS-1497: Terms of Use on patient's first login
* Added a proms/footer extension point
* Moved the Terms of Use link to the footer instead of the top appbar
* Adjusted the form pagination style to fit better with the footer

marta- committed:
CARDS-1574: Patient UI overhaul
Improved the look of step indicators (not dilled when pending, filled when done)

marta- committed:
CARDS-1573: Better alignment on the patient survey page

marta- committed:
CARDS-1574: Patient UI overhaul
Reduced some redundant calls to load the visit subject

marta- committed:
CARDS-1574: Patient UI overhaul
* Better looking appbar
* Fixed some JS console warnings (missing `key`)
* Improved the look of the welcome page (after patient login)
* Better positioning of Form header (fixing layout issue caused by introduction of Footer for 1497)

marta- committed:
CARDS-1497: Terms of Use on patient's first login
* Now the user needs to scroll all the way down and see the whole Terms of Use text before being able to click "Accept"

marta- commented:
Notable changes in the last 6 commits:
* The patient now must scroll all the way down and see the whole Terms of Use text before they can click accept
* Terms of Use link moved to the footer instead of the header
  * With the introduction of a footer extension point, we can easily add more links such as FAQ
  * The new footer caused significant layout issues to paginated forms, so the AppBar, Form header and Welcome Screen had to be refactored (the relevant commits are tagged with CARDS-1574 and CARDS-1573).

marta- reviewed (approved):
Approving for now. The code will need at least another reviewer.

sashaandjic reviewed (approved):
tested - awesome!

marta- commented:
Rebased on the latest `dev` which includes 1507.

acrowthe reviewed (approved):
Looks good: Only issue I noticed was that the progress bar shakes when the user is scrolled down slightly, but that seems like something that can be fixed in CARDS-1574
![ProgressBarShaking](https://user-images.githubusercontent.com/20056751/150018627-0235c1cb-6f79-41b9-a1ba-9d6a762b6e64.gif)

marta- commented:
> Only issue I noticed was that the progress bar shakes when the user is scrolled down slightly, but that seems like something that can be fixed in CARDS-1574

Ouch, that looks ugly! Let me see what I can do...

sashaandjic reviewed (approved):
re-tested - I do not see shaking progress bar ( tried to reproduce Firefox and Chrome)

marta- committed:
CARDS-1574: Patient UI overhaul
* Better styling of the survey title
* Keep the progress bar's position on the screen when collapsing toolbars on scrolling
* Progress bar moves forward when starting the first questionnaire

marta- commented:
The latest commit adds some minor AppBar improvements. The flickering issue is still not fixed.

sashaandjic commented:
Oh, I got the flickering finally. 
Uploading ePRO - Google Chrome 2022-01-19 04-16-43.mp4…

marta- committed:
CARDS-1574: Patient UI overhaul
Bugfix: the form title bar is flickering when scrolling down at certain heights because of the varying height of the container when the appbar collapses

marta- commented:
Flickering fixed in cec9373.

acrowthe commented:
Screen flickering fixed, looks good to me

sashaandjic reviewed (approved):
tested - no more flickering!

marta- commented:
Rebased on the latest `dev` which includes 1506.

veronikaslc committed:
CARDS-1576: If the terms of use change, patient users should be prompted to review and re-accept (#855)

CARDS-1576: If the terms of use change, patient users should be prompted to review and re-accept

Co-authored-by: veronikaslc <veronika.koltunova@sickkids.ca>
Co-authored-by: Marta Girdea <marta.girdea@gmail.com>

marta- commented:
Merged CARDS-1576 (_If the terms of use change, patient users should be prompted to review and re-accept_) in to this branch.

sdumitriu commented on proms-resources/frontend/src/main/frontend/src/ToUDialog.jsx:
```diff
@@ -0,0 +1,208 @@
+  }
+}));
+
+// Component that renders the Dialog width Terms of Use
```
```suggestion
// Component that renders the Dialog with Terms of Use
```

sdumitriu reviewed (changes requested):
Regression: In the proms login, pressing Enter to submit the login makes the app hang. One must use the mouse to click on the Submit button.

sdumitriu committed:
[misc] Typo fix

sdumitriu reviewed (approved):
The regression will be fixed in CARDS-1550.

marta- commented:
> Regression: In the proms login, pressing Enter to submit the login makes the app hang. One must use the mouse to click on the Submit button.

It's not a regression. It only happens when the patient information form doesn't exist yet, and is caused by the delay introduced by creating it. This will never happen in production when real patient identification is implemented. The PRs testing instructions clearly say the implementation assumes the form exists.
